### PR TITLE
Support UNF with sortable 

### PR DESCRIPTION
**Pull request by AvitalFineRedis**
Adding the ability to use UNF in Tag filds and Text fields.

### CODE REVIEW

File: tests/NRediSearch.Test/ClientTests/ClientTest.cs [hunk 1]
```diff
@@ -865,11 +867,34 @@ public void TestGetSortableTagField()
             Assert.Equal(4, cl.Search(new Query("hello")).TotalResults);
         }
 
+        [Fact]
+        public void TestGetTagFieldUnf() {
+            // Add version check
```
Comment (AvitalFineRedis) [inline tests/NRediSearch.Test/ClientTests/ClientTest.cs:872]: Note: after [this PR](https://github.com/StackExchange/StackExchange.Redis/pull/1841) will be merged, we need to make sure that this test is running only when the `RediSearch` version is greater than 2.0.12. The test will fail for previous versions.

File: tests/NRediSearch.Test/ClientTests/ClientTest.cs [hunk 2]
```diff
@@ -865,11 +867,34 @@ public void TestGetSortableTagField()
...
+
+            Client cl = GetClient();
+            sc = new Schema().AddSortableTextField("txt").AddSortableTextField("txt_unf", unf: true).
```
Comment (slorello89) [inline tests/NRediSearch.Test/ClientTests/ClientTest.cs:875]: syntax error, `sc` never declared.

File: src/NRediSearch/Schema.cs [hunk 3]
```diff
@@ -238,8 +245,9 @@ internal override void SerializeRedisArgs(List<object> args)
                     if (Sortable) args.Remove("SORTABLE");
                     args.Add("SEPARATOR".Literal());
                     args.Add(Separator);
-                    if (Sortable) args.Add("SORTABLE".Literal());
                 }
+                if (Sortable) args.Add("SORTABLE".Literal());
```
Comment (slorello89) [inline src/NRediSearch/Schema.cs:249]: The Base `SerializedRedisArgs` add SORTABLE to the args list, hence if you declare a Tag field sortable, you will have two `SORTABLE` args in the list of args sent to Redis, does not technically seem to be a syntax error but it ought to be, I suggest lifting the `UNF` add to the base class, and in this special case (where `SEPARATOR` really cares to be towards the back of the arguments list) remove `SORTBABLE` and `UNF` out of the list and add them back in if appropriate. 

> Note in my testing UNF is also a "valid" argument in numeric indices - appears to simply be inert as it's not applied inside of RediSearch when the indices are built, but it doesn't seem to mess with anything, so it should be harmless to put it in the base Serialize args and simply not reference it outside the tag and text field.

File: src/NRediSearch/Schema.cs [hunk 4]
```diff
@@ -86,6 +89,7 @@ internal override void SerializeRedisArgs(List<object> args)
                     args.Add(Weight);
                 }
                 if (NoStem) args.Add("NOSTEM".Literal());
+                if (Unf) args.Add("UNF".Literal());
```
Comment (slorello89) [inline src/NRediSearch/Schema.cs:92]: per my comment below this should be moved to the base class's `SerializeRedisArgs` method, also should be tightly tied with sortable as it's meant to be an option of the SORTABLE argument.

**Review by slorello89 — CHANGES_REQUESTED** [review]

**Pull request merged**